src/Aspire.Hosting.Azure/AzureBicepResource.cs

```diff
@@ -29,6 +38,23 @@ public AzureBicepResource(string name, string? templateFile = null, string? temp
         TemplateResourceName = templateResourceName;
 
         Annotations.Add(new ManifestPublishingCallbackAnnotation(WriteToManifest));
+
+        // Add pipeline step annotation to provision this bicep resource
+        Annotations.Add(new PipelineStepAnnotation((factoryContext) =>
+        {
+            // Initialize the provisioning task completion source during step creation
+            ProvisioningTaskCompletionSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+            
+            var provisionStep = new PipelineStep
+            {
+                Name = $"provision-{name}",
+                Action = async ctx => await ProvisionAzureBicepResourceAsync(ctx, this).ConfigureAwait(false)
```

Comment:
In the future this would call into a virtual method on `AzureBicepResource`. That would allow resources to write custom logic here.